.github/workflows/publish-to-branch.yaml

```diff
@@ -0,0 +1,28 @@
+name: Publish to branch
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+      - track/**
+
+jobs:
+  publish-charm:
+    name: Publish Charm
+    runs-on: ubuntu-latest
+    if: startsWith( github.head_ref, 'branch/' )
```

Comment:
We should confirm, but I think we're changing our philosophy here so that we **always** publish changes on github branchs to charmhub `latest/edge/branchname` branches, regardless of the github branch name (eg: we're no longer limiting "publish to branch" to only run on github branches named `branch/*`, but now we do it for all github branches)

Reply:
(Passerby comment):

[Branch name](https://stackoverflow.com/a/3651867/3516684) would probably need to be sanitized:

```shell
$ git checkout -b some/crazy/branch.with.periods
Switched to a new branch 'some/crazy/branch.with.periods'

# HOWEVER

$ charmcraft release karma-k8s --channel=edge/some/crazy/branch.with.periods ...
Store operation failed:
- api-error: Invalid channel: edge/some/crazy/branch.with.periods
```

Reply:
Good point.  I think this would best be caught in the `upload-charm` action - does that sound right to you @sed-i?

Reply:
I guess the answer is both:
- Own CI would be responsible for any custom name transformations.
- `upload-charm` would have to do the same generic sanitation for everything comes its way (or fail, depending on a flag passed?)